webapp/views.py

```diff
@@ -173,6 +177,73 @@ def post_build():
     return flask.render_template("core/build.html", **context)
 
 
+def notify_build():
+    """
+    An endpoint to trigger an update about a build event to be sent.
+    This will usually be triggered by a webhook from Launchpad
+    """
+
+    # Verify contents
+    signature = hmac.new(
+        flask.current_app.config["SECRET_KEY"].encode("utf-8"),
+        flask.request.data,
+        hashlib.sha1,
+    ).hexdigest()
+
+    if "X-Hub-Signature" not in flask.request.headers:
+        return "No X-Hub-Signature provided\n", 403
+
+    if not hmac.compare_digest(
+        signature, flask.request.headers["X-Hub-Signature"].split("=")[1]
+    ):
+        print(signature)
+        return "X-Hub-Signature does not match\n", 400
+
+    event_content = flask.request.json
+
+    build_url = (
+        "https://api.launchpad.net/devel" + event_content["livefs_build"]
+    )
+
+    launchpad = Launchpad(
+        username="image.build",
+        token=os.environ["LAUNCHPAD_TOKEN"],
+        secret=os.environ["LAUNCHPAD_SECRET"],
+        session=session,
+    )
+
+    build = launchpad.request(build_url).json()
+    email = build["metadata_override"]["_email"]
+    names = build["metadata_override"]["_name"].split(" ")
+    board = build["metadata_override"]["_board"]
+    snaps = ", ".join(build["metadata_override"]["extra_snaps"])
+    codename = build["distro_series_link"].split("/")[-1]
+    arch = build["distro_arch_series_link"].split("/")[-1]
+    build_link = build["web_link"]
+    status = build["buildstate"]
+
+    session.post(
+        "https://pages.ubuntu.com/index.php/leadCapture/save",
+        data={
+            "FirstName": " ".join(names[:-1]),
+            "LastName": names[-1] if len(names) > 1 else "",
+            "Email": email,
+            "formid": "3546",
+            "lpId": "2154",
+            "subId": "30",
+            "munchkinId": "066-EOV-335",
+            "imageBuilderVersion": codename,
+            "imageBuilderArchitecture": arch,
+            "imageBuilderBoard": board,
+            "imageBuilderSnaps": snaps,
+            "imageBuilderBuildlink": build_link,
```

Comment:
This is missing the imageBuilderDownloadlink, which is the link in the email the user clicks to download the image. Might be the same as the builld_link

Reply:
I mentioned this in the PR description:

> Marketo expects a `imageBuilderDownloadlink`, but I can't see any way to get this download URL from [the build object in the API](https://launchpad.net/+apidoc/devel.html#build).

I don't think we should use the `build_link`. That just takes you to the details page, which is not what you would expect of a "Download build" button. I don't know where we got the idea that we could get this URL, but we will either need to go back to Colin or remove the expectation for it to exist from Marketo and the email.

Reply:
This is the main piece of information so assumed the webhook would include that otherwise, the webhook is useless. We will need to go back to them, do you mind writing an email to them to raise this?

Reply:
This piece is now fixed